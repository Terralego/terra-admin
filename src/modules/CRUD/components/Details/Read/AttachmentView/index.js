import { withRouter } from 'react-router-dom';
import { withTranslation } from 'react-i18next';
import compose from '../../../../../../utils/compose';

import AttachmentView from './AttachmentView';

export default compose(
  withRouter,
  withTranslation(),
)(AttachmentView);
