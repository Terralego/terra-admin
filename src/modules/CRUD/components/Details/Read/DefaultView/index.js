import DefaultView from './DefaultView';

export default DefaultView;
