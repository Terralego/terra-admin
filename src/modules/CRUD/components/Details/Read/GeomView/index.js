import GeomView from './GeomView';

export default GeomView;
