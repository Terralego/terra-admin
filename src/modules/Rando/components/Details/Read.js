import React from 'react';
import { withRouter } from 'react-router';
import classnames from 'classnames';
import { withNamespaces } from 'react-i18next';

import Actions from './Actions';

const NO_FEATURE = 'rando.details.noFeature';

const Read = ({
  t,
  match: { params: { layer, id } },
  schema: { properties },
}) => {
  if (!properties) return null;
  const { name: { default: title } = {} } = properties;

  return (
    <div className="details">
      <div className="details__header">
        <h2 className="details__title">{title || t(NO_FEATURE)}</h2>
      </div>
      <div className="details__content">
        <h3 className="details__subtitle">{t('rando.details.informations')}</h3>
        <ul className="details__list">
          {Object.keys(properties).map(prop => (
            <li key={prop} className="details__list-item">
              <strong className="details__list-label">{properties[prop].title || prop}</strong>
              <span className={classnames('details__list-value', { 'details__list-value--empty': !properties[prop].default })}>
                {properties[prop].default || t(NO_FEATURE)}
              </span>
            </li>
          ))}
        </ul>
      </div>
      <Actions paramId={id} paramLayer={layer} displayUpdate displayDelete />
    </div>
  );
};
export default withRouter(withNamespaces()(Read));
