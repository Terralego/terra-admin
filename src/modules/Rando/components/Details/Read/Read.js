import React from 'react';
import { Redirect } from 'react-router-dom';
import classnames from 'classnames';

import { toast } from '../../../../../utils/toast';
import { generateURI } from '../../../config';
import Actions from '../Actions';

const NO_FEATURE = 'rando.details.noFeature';

const formattedProp = value => {
  if (typeof value === 'string') {
    // eslint-disable-next-line react/no-array-index-key
    return value.split('\n').map((item, i) => (<React.Fragment key={i}>{item} <br /></React.Fragment>));
  }
  if (Array.isArray(value)) {
    return value.join(', ');
  }
  return value;
};

const Read = ({
  t,
  match: { params: { layer, id } },
  schema: { properties = {} },
  displayViewFeature,
}) => {
  if (!displayViewFeature) {
    toast.displayError(t('rando.details.noAccess'));
    return (<Redirect to={generateURI('layer', { layer })} />);
  }

  const { name: { default: title } = {} } = properties;
  const hasProperties = !!Object.keys(properties).length;

  return (
    <div className="details">
      <div className="details__header">
        <h2 className="details__title">{title || t(NO_FEATURE)}</h2>
      </div>
      {hasProperties && (
        <div className="details__content">
          <h3 className="details__subtitle">{t('rando.details.informations')}</h3>
          <ul className="details__list">
            {Object.keys(properties).map(prop => (
              <li key={prop} className="details__list-item">
                <strong className="details__list-label">{properties[prop].title || prop}</strong>
                <span className={classnames('details__list-value', { 'details__list-value--empty': !properties[prop].default })}>
                  {formattedProp(properties[prop].default || t(NO_FEATURE))}
                </span>
              </li>
            ))}
          </ul>
        </div>
      )}
      <Actions paramId={id} paramLayer={layer} displayUpdate displayDelete />
    </div>
  );
};
export default Read;
