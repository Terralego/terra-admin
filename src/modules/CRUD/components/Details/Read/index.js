
import { withRouter } from 'react-router-dom';
import { withTranslation } from 'react-i18next';
import { connectAuthProvider } from '@terralego/core/modules/Auth';
import compose from '../../../../../utils/compose';

import Read from './Read';

const authProviderGetter = ({
  authenticated,
  user,
}) => {
  const permissions = authenticated ? user.permissions : [];
  return {
    displayViewFeature: permissions.includes('can_view_feature'),
  };
};

export default compose(
  withRouter,
  connectAuthProvider(authProviderGetter),
  withTranslation(),
)(Read);
