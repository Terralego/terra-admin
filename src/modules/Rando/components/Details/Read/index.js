
import { withRouter } from 'react-router';
import { withNamespaces } from 'react-i18next';
import { connectAuthProvider } from '@terralego/core/modules/Auth';

import Read from './Read';

export default connectAuthProvider(({
  user,
}) => {
  const { permissions = [] } = user;
  return {
    displayViewFeature: permissions.includes('terra.view_feature'),
  };
})(
  withRouter(
    withNamespaces()(
      Read,
    ),
  ),
);
