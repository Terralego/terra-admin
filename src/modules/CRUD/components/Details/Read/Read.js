import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';

import { toast } from '../../../../../utils/toast';
import { generateURI } from '../../../config';
import Header from '../Header';
import Menu from '../Menu';
import DefaultView from './DefaultView';
import GeomView from './GeomView';
import AttachmentView from './AttachmentView';

const Read = ({
  t,
  match: { params: { layer, section = 'default' } },
  displayViewFeature,
  feature: {
    title = t('CRUD.details.noFeature'),
    documents,
    geometry_files: geomFiles,
    geometries,
    display_properties: displayProperties,
    properties,
  },
  ...rest
}) => {
  if (!displayViewFeature) {
    toast.displayError(t('CRUD.details.noAccess'));
    return (<Redirect to={generateURI('layer', { layer })} />);
  }

  const getView = () => {
    switch (section) {
      case 'attachments':
        return <AttachmentView />;
      case 'geometries':
        return <GeomView geometries={geometries} {...rest} />;
      default:
        return <DefaultView displayProperties={displayProperties} properties={properties} />;
    }
  };

  return (
    <div className="details">
      <Header
        generatedFiles={documents}
        geomFiles={geomFiles}
        title={title}
      />
      <Menu section={section} />
      <div className="details__content">
        {getView()}
      </div>
    </div>
  );
};

Read.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      layer: PropTypes.string,
      section: PropTypes.string,
    }),
  }),
  displayViewFeature: PropTypes.bool,
  feature: PropTypes.shape({
    title: PropTypes.string,
    documents: PropTypes.array,
    display_properties: PropTypes.shape({}),
  }),
  t: PropTypes.func,
};

Read.defaultProps = {
  match: {
    params: {
      layer: undefined,
      section: undefined,
    },
  },
  displayViewFeature: true,
  feature: {
    title: '',
    documents: [],
    display_properties: {},
  },
  t: text => text,
};

export default Read;
