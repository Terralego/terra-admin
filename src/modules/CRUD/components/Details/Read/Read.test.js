import React from 'react';
import renderer from 'react-test-renderer';

import Read from './Read';

jest.mock('react-router-dom', () => ({
  Redirect: () => <div>Error because Redirect</div>,
}));

jest.mock('../../../../../utils/toast', () => ({
  toast: {
    displayError: children => <div>{children}</div>,
  },
}));

jest.mock('../../../config', () => ({
  generateURI: jest.fn(),
}));

jest.mock('../Header', () => props => <div {...props}>Header</div>);
jest.mock('../Menu', () => props => <div {...props}>Menu</div>);
jest.mock('./DefaultView', () => props => <div {...props}>Default view</div>);
jest.mock('./AttachmentView', () => props => <div {...props}>Attachment view</div>);
jest.mock('./GeomView', () => props => <div {...props}>Geom view</div>);

const props = {
  t: text => text,
  match: { params: { layer: 'layerFoo' } },
  displayViewFeature: true,
  feature: {
    title: 'Title of the feature',
    display_properties: {
      foo: 'foo',
    },
    new_display_properties: {
      foo: 'foo',
    },
    documents: [{
      file1: 'file1',
      file2: 'file2',
    }],
  },
};

it('should redirect', () => {
  const tree = renderer.create((
    <Read
      {...props}
      displayViewFeature={false}
    />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});


it('should render defaultView correctly', () => {
  const tree = renderer.create((
    <Read
      {...props}
    />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render AttachmentView correctly', () => {
  const tree = renderer.create((
    <Read
      {...props}
      match={{ params: { layer: 'layerFoo', section: 'attachmentFiles' } }}
    />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
