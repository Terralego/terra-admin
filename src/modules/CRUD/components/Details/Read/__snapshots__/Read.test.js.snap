// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should redirect 1`] = `
<div>
  Error because Redirect
</div>
`;

exports[`should render AttachmentView correctly 1`] = `
<div
  className="details"
>
  <div
    generatedFiles={
      Array [
        Object {
          "file1": "file1",
          "file2": "file2",
        },
      ]
    }
    title="Title of the feature"
  >
    Header
  </div>
  <div
    section="attachmentFiles"
  >
    Menu
  </div>
  <div
    className="details__content"
  >
    <div
      displayProperties={
        Object {
          "foo": "foo",
        }
      }
    >
      Default view
    </div>
  </div>
</div>
`;

exports[`should render defaultView correctly 1`] = `
<div
  className="details"
>
  <div
    generatedFiles={
      Array [
        Object {
          "file1": "file1",
          "file2": "file2",
        },
      ]
    }
    title="Title of the feature"
  >
    Header
  </div>
  <div
    section="default"
  >
    Menu
  </div>
  <div
    className="details__content"
  >
    <div
      displayProperties={
        Object {
          "foo": "foo",
        }
      }
    >
      Default view
    </div>
  </div>
</div>
`;
